src/Frame.cpp

```diff
@@ -1,251 +1,626 @@
+
+	auto a = cross(rij, rjk);
+	auto b = cross(rjk, rkm);
+	return atan2(rjk.norm() * dot(b, rij), dot(a, b));
 }
 
 double Frame::out_of_plane(size_t i, size_t j, size_t k, size_t m) const {
-    if (i >= size() || j >= size() || k >= size() || m >= size()) {
-        throw out_of_bounds(
-            "out of bounds atomic index in `Frame::out_of_plane`: we have {} "
-            "atoms, but the index are {}, {}, {}, and {}",
-            size(), i, j, k, m
-        );
-    }
-
-    auto rji = cell_.wrap(positions_[j] - positions_[i]);
-    auto rik = cell_.wrap(positions_[i] - positions_[k]);
-    auto rim = cell_.wrap(positions_[i] - positions_[m]);
-
-    auto n = cross(rik, rim);
-    auto n_norm = n.norm();
-    if (n_norm < 1e-12) {
-        // if i, k, and m are colinear, then j is always inside the plane
-        return 0;
-    } else {
-        return dot(rji, n) / n_norm;
-    }
+	if (i >= size() || j >= size() || k >= size() || m >= size()) {
+		throw out_of_bounds(
+			"out of bounds atomic index in `Frame::out_of_plane`: we have {} "
+			"atoms, but the index are {}, {}, {}, and {}",
+			size(), i, j, k, m
+		);
+	}
+
+	auto rji = cell_.wrap(positions_[j] - positions_[i]);
+	auto rik = cell_.wrap(positions_[i] - positions_[k]);
+	auto rim = cell_.wrap(positions_[i] - positions_[m]);
+
+	auto n = cross(rik, rim);
+	auto n_norm = n.norm();
+	if (n_norm < 1e-12) {
+		// if i, k, and m are colinear, then j is always inside the plane
+		return 0;
+	}
+	else {
+		return dot(rji, n) / n_norm;
+	}
 }
 
 const std::unordered_map<std::string, double> BOND_GUESSING_RADII = {
-    {"H", 1.0},
-    {"C", 1.5},
-    {"O", 1.3},
-    {"N", 1.4},
-    {"S", 1.9},
-    {"F", 1.2},
+	{"H", 1.0},
+	{"C", 1.5},
+	{"O", 1.3},
+	{"N", 1.4},
+	{"S", 1.9},
+	{"F", 1.2},
 };
 
 optional<double> guess_bonds_radius(const Atom& atom) {
-    const auto& type = atom.type();
-    auto it = BOND_GUESSING_RADII.find(type);
-    if (it != BOND_GUESSING_RADII.end()) {
-        // allow configuration file to override data from BOND_GUESSING_RADII
-        auto user_config = Configuration::atom_data(type);
-        if (user_config && user_config->vdw_radius) {
-            return user_config->vdw_radius.value();
-        } else {
-            return it->second;
-        }
-    } else {
-        // default to chemfiles provided atom type
-        return atom.vdw_radius();
-    }
+	const auto& type = atom.type();
+	auto it = BOND_GUESSING_RADII.find(type);
+	if (it != BOND_GUESSING_RADII.end()) {
+		// allow configuration file to override data from BOND_GUESSING_RADII
+		auto user_config = Configuration::atom_data(type);
+		if (user_config && user_config->vdw_radius) {
+			return user_config->vdw_radius.value();
+		}
+		else {
+			return it->second;
+		}
+	}
+	else {
+		// default to chemfiles provided atom type
+		return atom.vdw_radius();
+	}
+}
+
+int Frame::generateBonds(std::vector<int>* result, std::vector<Vector3D>* positions, int totalCells, int** cells,
```

Comment:
Pointers is not the usual tool to reach for in C++, it is more of a C thing to do. For example, it seems that `result` is used as an output parameter, the usual way of doing this in C++ would be to pass a reference: `std::vector<int>& result`. Same for positions, cells and all the pointer parameters.

Reply:
You are right. Even if I make sure to initialize the result variable before passing it to the function and I know the other variables exist but it is better to pass a reference instead. It will be changed in the next commit